apps/api-documenter/src/documenters/MarkdownDocumenter.ts

```diff
@@ -0,0 +1,694 @@
+        case ApiItemKind.MethodSignature: {
+          methodsTable.addRow(
+            new DocTableRow({ configuration }, [
+              this._createTitleCell(apiMember),
+              this._createDescriptionCell(apiMember)
+            ])
+          );
+
+          this._writeApiItemPage(apiMember);
+          break;
+        }
+        case ApiItemKind.PropertySignature: {
+
+          if ((apiMember as ApiPropertyItem).isEventProperty) {
+            eventsTable.addRow(
+              new DocTableRow({ configuration }, [
+                this._createTitleCell(apiMember),
+                this._createResultTypeCell(apiMember),
+                this._createDescriptionCell(apiMember)
+              ])
+            );
+          } else {
+            propertiesTable.addRow(
+              new DocTableRow({ configuration }, [
+                this._createTitleCell(apiMember),
+                this._createResultTypeCell(apiMember),
+                this._createDescriptionCell(apiMember)
+              ])
+            );
+          }
+
+          this._writeApiItemPage(apiMember);
+          break;
+        }
+
+      }
+    }
+
+    if (eventsTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Events' }));
+      output.appendNode(eventsTable);
+    }
+
+    if (propertiesTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Properties' }));
+      output.appendNode(propertiesTable);
+    }
+
+    if (methodsTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Methods' }));
+      output.appendNode(methodsTable);
+    }
+  }
+
+  /**
+   * GENERATE PAGE: FUNCTION-LIKE
+   */
+  private _writeFunctionLikeTables(output: DocSection, apiFunctionLike: ApiFunctionLikeMixin): void {
+    const configuration: TSDocConfiguration = this._tsdocConfiguration;
+
+    const parametersTable: DocTable = new DocTable({ configuration,
+      headerTitles: [ 'Parameter', 'Type', 'Description' ]
+    });
+
+    for (const apiParameter of apiFunctionLike.parameters) {
+      const parameterDescription: DocSection = new DocSection({ configuration } );
+      if (apiParameter.tsdocParamBlock) {
+        this._appendSection(parameterDescription, apiParameter.tsdocParamBlock.content);
+      }
+
+      parametersTable.addRow(
+        new DocTableRow({ configuration }, [
+          new DocTableCell({configuration}, [
+            new DocParagraph({ configuration }, [
+              new DocPlainText({ configuration, text: apiParameter.name })
+            ])
+          ]),
+          new DocTableCell({configuration}, [
+            new DocParagraph({ configuration }, [
+              new DocCodeSpan({ configuration, code: apiParameter.resultTypeSignature })
+            ])
+          ]),
+          new DocTableCell({configuration}, parameterDescription.nodes)
+        ])
```

Comment:
What's the difference between using this and using `this._createDescriptionCell`?

Reply:
`_createDescriptionCell()` is for ApiDocumentedItems that have a TSDoc comment attached to them.  `ApiParameter` (somewhat counterintuitively) does not have its own TSDoc comment.  Instead, it gets a little section from the `@param` tag of the function-like thing that it belongs to.

I'll add some comments to make this more clear.  Maybe we can also tighten up the typings so you can't accidentally pass the wrong thing.

---
In reply to: [236365223](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236365223) [](ancestors = 236365223)